src/qtmozembed/declarativewebpage.cpp

```diff
@@ -370,9 +384,11 @@ void DeclarativeWebPage::updateViewMargins()
         qreal threshold = qMax(m_fullScreenHeight * 1.5f, (m_fullScreenHeight + (m_toolbarHeight*2)));
         if (contentHeight() < threshold) {
             margins.setBottom(m_toolbarHeight);
+            chromeVisible = true;
         }
     }
 
+    forceChrome(chromeVisible);
```

Comment:
forceChrome disables the gesture to hide/show toolbar when panning content. When content height is big enough, we can let the gesture to work other wise it's disabled.